daemon/internal/newrelic/integration/test.go

```diff
@@ -456,6 +458,123 @@ func (t *Test) compareSpanEventsLike(harvest *newrelic.Harvest) {
 	}
 }
 
+// Handles EXPECT_METRICS_EXIST
+func (t *Test) compareMetricsExist(harvest *newrelic.Harvest) {
+	for _, spec := range strings.Split(strings.TrimSpace(string(t.subEnvVars(t.metricsExist))), "\n") {
+		var err error
+		var count int64
+
+		fields := strings.Split(spec, ",")
+		name := fields[0]
+		name = strings.TrimSpace(name)
+
+		// -1 means just test for existence not an exact count
+		count = -1
+		if len(fields) == 2 {
+			countStr := strings.TrimSpace(fields[1])
+			if len(countStr) > 0 {
+				// is it "??" - with or without quotes
+				match, _ := regexp.MatchString("^\"*\\?\\?\"*$", countStr)
+				if match {
+					count = -1
+				} else {
+					count, err = strconv.ParseInt(countStr, 10, 64)
+					if nil != err {
+						t.Fail(fmt.Errorf("EXPECT_METRICS_EXIST has unparsable count: %s", spec))
+					}
+				}
```

Comment:
Since `count` is already set to `-1` above, it feels like we can optimize this nested if block.
```suggestion
				if !match {
					count, err = strconv.ParseInt(countStr, 10, 64)
					if nil != err {
						t.Fail(fmt.Errorf("EXPECT_METRICS_EXIST has unparsable count: %s", spec))
					}
				}
```
This cuts out the need for the `else` statement while preserving the existing logic.